src/Okta.Idx.Sdk/OktaVerify/OktaVerifyAuthenticationOptions.cs

```diff
@@ -0,0 +1,253 @@
+        /// </summary>
+        /// <returns>OktaVerifyPollResponse.</returns>
+        /// <exception cref="ArgumentException">If the challenge authenticator is not valid.</exception>
+        public async Task<OktaVerifyPollResponse> PollOnceAsync()
+        {
+            if (ChallengePollRemediationOption.Name != RemediationType.ChallengePoll)
+            {
+                throw new ArgumentException($"Expected remediation option of type '{RemediationType.ChallengePoll}', the specified remediation option is of type {ChallengePollRemediationOption.Name}.");
+            }
+
+            IdxRequestPayload requestPayload = new IdxRequestPayload
+            {
+                StateHandle = StateHandle,
+            };
+
+            try
+            {
+                var challengeResponse = await ChallengePollRemediationOption.ProceedAsync(requestPayload);
+                bool continuePolling = challengeResponse.ContainsRemediationOption(RemediationType.ChallengePoll, out IRemediationOption challengePollRemediationOption);
+
+                if (challengeResponse.SuccessWithInteractionCode != null)
+                {
+                    TokenInfo = await challengeResponse.SuccessWithInteractionCode.ExchangeCodeAsync(IdxContext);
+                    return new OktaVerifyPollResponse
+                    {
+                        AuthenticationStatus = AuthenticationStatus.Success,
+                        TokenInfo = TokenInfo,
+                        ContinuePolling = continuePolling,
+                    };
+                }
+
+                if (challengeResponse.ContainsRemediationOption(RemediationType.SelectAuthenticatorAuthenticate))
+                {
+                    return new OktaVerifyPollResponse
+                    {
+                        AuthenticationStatus = AuthenticationStatus.AwaitingChallengeAuthenticatorSelection,
+                        ContinuePolling = continuePolling,
+                    };
+                }
+
+                if (challengeResponse.ContainsRemediationOption(RemediationType.SelectAuthenticatorEnroll))
+                {
+                    return new OktaVerifyPollResponse
+                    {
+                        AuthenticationStatus = AuthenticationStatus.AwaitingAuthenticatorEnrollment,
+                        ContinuePolling = continuePolling,
+                    };
+                }
+
+                throw new UnexpectedRemediationException(
+                    new List<string>
+                    {
+                                    RemediationType.SelectAuthenticatorAuthenticate,
+                                    RemediationType.SelectAuthenticatorEnroll,
+                    },
+                    challengeResponse);
+            }
+            catch (Exception ex)
+            {
+                return new OktaVerifyPollResponse
+                {
+                    AuthenticationStatus = AuthenticationStatus.Exception,
```

Comment:
We should follow the existing design and throw an exception with whatever error the server returns.